src/Counter.jsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+
+export class Counter extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {seconds: props.seconds}
+    }
+
+    componentDidMount() {
+        this.tickTimerId = setInterval(() => this.tick(), 1000)
+        setTimeout(() => this.turnOffTimer(), 11000)
```

Comment:
Плохая идея - основываться на том, что через 11 секунд нужно выключить таймер. А потом мы захотим, чтобы значение счётчика изменялось каждые две секунды вместо одной, и тогда счёт закончится, не дойдя до нужного числа. Предлагаю в методе tick() проверять: если мы дошли до нужной границы, то вызываем turnOffTimer(). Кроме того, просьба поправить: если счёт начинается в 0, то он должен закончиться на 10, если начинается на 10, то должен закончиться на 20.

Reply:
Про тик хорошо, поправлю.
У меня так и есть: счет начинается в 0, а заканчивается в 10 у одного таймера и в 10 начинается, в 20 заканчивается у другого

Reply:
@dshuran поправил)

Reply:
You can use functional components to implement this.

```
function Counter({ name, startSeconds, tickInterval, endSeconds }) {
  const [seconds, setSeconds] = useState(startSeconds);

  useEffect(() => {
    let currentSeconds = startSeconds;

    const intervalId = setInterval(() => {
      if (currentSeconds >= endSeconds) {
        clearInterval(intervalId);
      }
      currentSeconds++;
      setSeconds(currentSeconds);
    }, []);

    return () => {
      clearInterval(intervalId);
    };
  }, [startSeconds, tickInterval, endSeconds]);

  return (
    <div className="counter">
      <p className="centeredText">This is {this.props.name} counter!</p>
      <p className="centeredText bigText">{this.state.seconds}</p>
    </div>
  );
}
```

Reply:
Also it allows us to properly handle cases when props are changing.

Reply:
@whiteand Hello, Andrew :)
Actually, we don't use hooks for now because this is the react course for beginners whereas I consider hooks as a more complex tool. Also it doesn't make code more readable. Moreover, use need to specify all this properties that don't change as a second argument for useEffect hook. I find this syntax confusing for beginners. 

You can write me in Telegram (@dshuran) and I'll share with you some materials from course, so you can get more complete understanding